smtr/br_rj_riodejaneiro_viagens/viagens_realizadas.sql

```diff
@@ -0,0 +1,75 @@
+-- get start/end of trips, detects vehicles coming in the buffer at each boundary point
+with 
+t as (
+    SELECT 
+        *,
+        string_agg(status_viagem,"") over (
+            PARTITION BY id_veiculo, shape_id
+            ORDER BY id_veiculo, shape_id, timestamp_gps
+            ROWS BETWEEN CURRENT ROW AND 1 FOLLOWING) = 'startmiddle' starts,
+        string_agg(status_viagem,"") over (
+            PARTITION BY id_veiculo, shape_id
+            ORDER BY id_veiculo, shape_id, timestamp_gps
+            ROWS BETWEEN 1 PRECEDING AND CURRENT ROW) = 'middleend' ends
+    FROM `rj-smtr-dev.br_rj_riodejaneiro_onibus_gps.aux_registros_status_viagem`
+    WHERE 
+        data BETWEEN DATE('2022-02-01 10:10:10') AND DATE('2022-03-01 10:10:10')
+    -- ORDER BY shape_id, timestamp_gps
```

Comment:
```suggestion
    ORDER BY shape_id, timestamp_gps
```

Precisa estar ordenado para a próxima subquery, de identificação do `datetime_partida` e `datetime_chegada`